cloud_governance/cloud_resource_orchestration/aws/ec2/cost_over_usage.py

```diff
@@ -0,0 +1,167 @@
+        self.current_start_date = self.current_end_date - timedelta(days=self.__cro_duration_range)
+        self.__public_cloud_name = self.__environment_variables_dict.get('PUBLIC_CLOUD_NAME')
+        self.__ce_operations = CostExplorerOperations()
+        self.es_operations = ElasticSearchOperations(es_host=self.__es_host, es_port=self.__es_port)
+
+    def get_cost_based_on_tag(self, start_date: str, end_date: str, tag_name: str, extra_filters: dict = None, extra_operation: str = 'And', granularity: str = None):
+        """
+        This method gives the cost based on the tag_name
+        :param granularity:
+        :param extra_operation: default, And
+        :param extra_filters:
+        :param tag_name:
+        :param start_date:
+        :param end_date:
+        :return:
+        """
+        remove_savings_cost = {  # removed the savings plan usage from the user costs
+            'Not': {
+                'Dimensions': {
+                    'Key': 'RECORD_TYPE',
+                    'Values': ['SavingsPlanRecurringFee', 'SavingsPlanNegation', 'SavingsPlanCoveredUsage']
+                }
+            }
+        }
+        Filters = remove_savings_cost
+        if extra_filters:
+            Filters = {
+                extra_operation: [
+                    extra_filters,
+                    remove_savings_cost
+                ]
+            }
+        results_by_time = self.__ce_operations.get_cost_by_tags(start_date=start_date, end_date=end_date, tag=tag_name, Filter=Filters, granularity=granularity)['ResultsByTime']
+        response = self.__ce_operations.get_filter_data(ce_data=results_by_time, tag_name=tag_name)
+        return response
+
+    def get_monthly_user_es_cost_data(self, tag_name: str = 'User', start_date: datetime = None, end_date: datetime = None, extra_matches: dict = None, granularity: str = 'MONTHLY', extra_operation: str = 'And'):
+        """
+        This method get the user cost from the es-data
+        :param tag_name: by default User
+        :param start_date:
+        :param end_date:
+        :param extra_matches:
+        :param granularity: by default MONTHLY
+        :param extra_operation:
+        :return:
+        """
+        if not start_date and not end_date:
+            end_date = self.current_end_date.date()
+            start_date = self.current_start_date.replace(day=1).date()
+        elif not start_date:
+            start_date = self.current_start_date.date()
+            end_date = end_date.date()
+        else:
+            if not end_date:
+                end_date = self.current_end_date.date()
+                start_date = start_date.date()
+            start_date = start_date.date()
+            end_date = end_date.date()
+        return self.get_cost_based_on_tag(start_date=str(start_date), end_date=str(end_date), tag_name=tag_name, granularity=granularity, extra_filters=extra_matches, extra_operation=extra_operation)
+
+    def get_cost_over_usage_users(self):
```

Comment:
I thought we discuss to fetch user cost direct from cost explorer (boto3) and not from elastic_search in case of es will be down